src/include/sof/audio/audio_stream.h

```diff
@@ -46,17 +46,20 @@ struct sof_audio_stream_params {
 	uint32_t rate;		/**< Number of data frames per second [Hz] */
 	uint16_t channels;	/**< Number of samples in each frame */
 
-	/** alignment limit of stream copy, this value indicates how many
-	 * integer frames which can meet both byte align and frame align
-	 * requirement. it should be set in component prepare or param functions
+	/**
+	 * align_frame_cnt indicates minimum frames can meet both bytes align and
+	 * frames align requirement. Eg: for a stereo 16-bit stream, the processing
+	 * data must be 8 bytes align as well as 3 frames align to meet the xtensa
+	 * intrinsic and algorithm requirement, then the align_frame_cnt is 6.
```

Comment:
Would 16bit stereo not be 4 bytes aligned instead of 8 ?

Reply:
the multi-way load/store xtensa intrinsic needs to be 8-byte aligned.

Reply:
> the multi-way load/store xtensa intrinsic needs to be 8-byte aligned.

ok, so the instrinsic requirement is 8 bytes (2 frames) and the algo requiremnet is 3 frames and hence the common factor processing block size is 6 frames.. You need to break it down as follows so it's easier to follow.

e.g. Consider an algorithm that processes in blocks of 3 frames configured to prcess 16bit stereo using xtensa HiFi3 SIMD. Therefore with 16 bit stereo we have a frame size of 4 bytes, and SIMD intrinsic requirment of 2 frames for HiFi X and a an algorithim requirment of 3 frames. Hence the common processing block size has to align with frame(1), intrinsic(2) and algorithm (3) giving us an optimum processing block size of 6 frames.